simulator/simulator.cpp

```diff
@@ -64,8 +65,10 @@ int main(int argc, char* argv[]) {
             Field_Dimensions::Current_Dimensions =
                 Field_Dimensions::Single_Field_Dimensions * scaling;
         } else if (strcmp(argv[i], "--timeout") == 0) {
-            timeoutsimulator = RJ::Seconds(stoi(argv[i + 1]));
-            i++;
+            if (i < argc - 1) {
+                timeoutsimulator = RJ::Seconds(stoi(argv[i + 1]));
+                i++;
+            }
```

Comment:
In the case that this if statement dosen't run, we should fail and print the usage. There's a usage function above that should do thiss for us!